backend/include/Server/controller/Authentication/AuthenticationController.hpp

```diff
@@ -224,27 +224,30 @@ class AuthenticationController : public oatpp::web::server::api::ApiController {
         m_logger->log->debug("/GET Logout");
 
         std::string cookie = ocookie.getValue("");
+        if (cookie.length() == 0)
+            return setupResponseWithCookieHeaders(createResponse(Status::CODE_401, "Cookie Missing"));
         auto spos = cookie.find("auth=");
-        std::string cdata = cookie.substr(spos + 5, cookie.find(';', spos) - spos - 5);
-        auto usrc = getCookieAuthData(decrypt(decodeBase64(cdata)));
+        if (spos < 0 || spos > cookie.length())
+            return setupResponseWithCookieHeaders(createResponse(Status::CODE_401, "Cookie Missing"));
+
+        auto usrc = getCookieAuthData(decrypt(decodeBase64(cookie.substr(spos + 5, cookie.find(';', spos) - spos - 5))));
         auto usr = m_database->getUser(usrc.puid);
 
         if (usr)
         {
             if (usr->sid.getValue("") == usrc.sid&& usr->sid != "")
             {
                 m_database->updateStringField("email", usr->email, "sid", "");
-                auto res = createResponse(Status::CODE_200, "user logged out");
-                res->putHeader("Access-Control-Allow-Credentials", "true");
+                auto res = setupResponseWithCookieHeaders(createResponse(Status::CODE_200, "user logged out"));
+                auto c = fmt::format("auth=;Max-Age=0;Domain={0};Path=/;HttpOnly{1}", ZWOO_DOMAIN, USE_SSL ? ";Secure" : "");
+                res->putHeader("Set-Cookie", c);
                 return res;
             }
             else
                 return createResponse(Status::CODE_401, "session id not matching!");
```

Comment:
z.B. Hier fehlen noch die anderen Headers

Reply:
welche anderen?

Reply:
Theoretisch alle, ich schaue es mir dann aber nochmal in Ruhe an.

Reply:
werden sie doch in Zeile 241 mit setupResponseWithCookieHeaders

Reply:
habs für lesbarkeit zum return statement getan